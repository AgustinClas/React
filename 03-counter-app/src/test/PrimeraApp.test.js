import React from "react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';

describe('Pruebas en PrimeraApp', () => {

    //No funciona porque cambiamos la configuracion de setupTest.js
    //con la config actual no existe toBeInTheDocument()

    /* test('debe mostrar el mensaje "Hola Agustin" ', () => {
        
        const saludo = "Hola Agustin";

        const { getByText } = render( <PrimeraApp saludo = { saludo }/> );

       expect(getByText( saludo )).toBeInTheDocument();
    }) */

    test('debe mostrar <PrimeraApp /> correctamente', () => {
        
        const wrapper = shallow(<PrimeraApp saludo="Hola Agustin"/>);

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe mostrar el subtitulo enviado ', () => {
        const saludo = 'Hola Agustin';
        const subtitulo = "Como andas?";
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subtitulo );
    })
    
    
})
