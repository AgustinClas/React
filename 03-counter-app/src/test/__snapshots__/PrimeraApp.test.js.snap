// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Agustin
  </h1>
  <p>
    Como va?
  </p>
</Fragment>
`;
